### DISCUSSION TITLE
TypeError: no implicit conversion of false into String

### DISCUSSION THREAD

**Discussion by derekPrivate** [Homebrew/discussions#4279] [Linux]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 4d0154fa7926e8a7b06e9b074dd45957e820407c
Last commit: 26 hours ago
Core tap JSON: 28 Feb 15:05 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 1
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: single-core 64-bit haswell
Clang: N/A
Git: 2.38.3 => /usr/bin/git
Curl: 7.61.1 => /usr/bin/curl
Kernel: Linux 4.14.281-144.502.amzn1.x86_64 x86_64 GNU/Linux
OS: Unknown
Host glibc: 2.17
/usr/bin/gcc: 4.8.5
/usr/bin/ruby: 2.0.0
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Trying to 
    brew install gcc
on Amazon Linux 1 fails. Using --verbose --debug shows

==> Postinstalling gcc
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/brew.rb (Formulary::FromPathLoader): loading /home/linuxbrew/.linuxbrew/opt/gcc/.brew/gcc.rb
Warning: The post-install step did not complete successfully
You can try again using:
  brew postinstall gcc
==> An exception occurred within a child process:
  TypeError: no implicit conversion of false into String

/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils/popen.rb:47:in `initialize'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils/popen.rb:47:in `popen'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils/popen.rb:47:in `popen'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils/popen.rb:9:in `popen_read'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils/popen.rb:16:in `safe_popen_read'
/home/linuxbrew/.linuxbrew/opt/gcc/.brew/gcc.rb:151:in `post_install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1160:in `block (2 levels) in run_post_install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:968:in `with_logging'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1159:in `block in run_post_install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils.rb:606:in `with_env'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1148:in `run_post_install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/postinstall.rb:24:in `<main>'

How can I fix?

**Comment by carlocab** [comment]
Can you report this at https://github.com/Homebrew/homebrew-core/issues?

As a workaround for now, try
```sh
export HOMEBREW_NO_INSTALL_FROM_API=1
brew tap homebrew/core
brew update
brew reinstall gcc
```

**Comment by derekPrivate** [comment]
Thanks for the tip

When I do that, no error are reported, BUT the homebew gcc does not seem to be installed (at least, it is not on the PATH). 
Running ‘which gcc’ is /usr/bin/gcc and reports gcc version 4.8.5 20150623 (Red Hat 4.8.5-28) (GCC) 

—
Derek Morris
***@***.***

**Reply by carlocab** [reply]
Try `which gcc-12`.

**Comment by derekPrivate** [comment]
/home/linuxbrew/.linuxbrew/bin/gcc-12

But gcc is not linked. Is that expected?

—
Derek Morris
***@***.***

**Reply by carlocab** [reply]
Yes. We install GCC with a version suffix, so you must call it as `gcc-12` or `g++-12`.
